pkg/kbld/image/factory.go

```diff
@@ -43,6 +50,10 @@ func (f Factory) New(url string) Image {
 	}
 
 	if srcConf, found := f.shouldBuild(url); found {
+		if !f.opts.AllowedToBuild {
+			return NewErrImage(fmt.Errorf("Building of images is disallowed (tried to build '%s')", url))
```

Comment:
🤔  What are your thoughts on explicitly naming the mechanism that causes this to occur?  Like...

"Building of images is disallowed (tried to build '%s' because a source was configured for it)"

Being explicit about causes makes the error message accessible/actionable for even casual users.

I see one concern: that this function should not _assume_ why the build was attempted. So then, `shouldBuild()` should return a `string` that's the reason for why the build should occur...

```go
	if srcConf, found, reason := f.shouldBuild(url); found {
		if !f.opts.AllowedToBuild {
			return NewErrImage(fmt.Errorf("Building of images is disallowed (tried to build '%s' because %s)", url, reason))
```

Reply:
`reason = "a source was configured for it"` for now would effectively be hard coding. i would side on less work and just hard code into the error directly for now. unless you are thinking there are other reason that exist...

Reply:
Going further takes us into a "purist" mindset. We're more pragmatic than that, here.

Let's hardcode this reason... it's — in fact — not likely to change.